### PR TITLE
Certificates implemented consistency changes besides pollers

### LINKED ISSUE #9112: Certificates consistency fixes
- [x] Type annotation in merge_certificate of x509_certificates from List[bytearray] -> Iterable[bytearray]
- [x] Type annotation of error in CertificateOperation from models.Error -> CertificateOperationError
- [x] Get rid of expires_on property of CertificatePolicy
- [x] (Optional) Make provider required in the CertificateIssuer constructor
- [x] Get rid of vault_url property in IssuerProperties
- [x] Add comments about how the create_certificate pollers will throw if you don't have get/permissions

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #9112

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_client.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def begin_create_certificate(
         """Creates a new certificate.
 
         If this is the first version, the certificate resource is created. This
-        operation requires the certificates/create permission.
+        operation requires the certificates/create permission. The poller requires the
+        certificates/get permission, otherwise it will throw a 403.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_client.py]: ~throw a 403~ raise :class:...HttpResponseError...

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_polling.py [hunk 2]
```diff
@@ -92,6 +92,9 @@ def _update_status(self):
         except ResourceNotFoundError:
             pass
         except HttpResponseError as e:
+            # We are exploiting a KV bug here. If we are polling on get_deleted_* and we don't
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_polling.py]: Suggest removing the first sentence.

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py [hunk 3]
```diff
@@ -55,7 +55,8 @@ class CertificateClient(AsyncKeyVaultClientBase):
         """Creates a new certificate.
 
         If this is the first version, the certificate resource is created. This
-        operation requires the certificates/create permission.
+        operation requires the certificates/create permission. The poller requires the
+        certificates/get permission, otherwise it will throw a 403.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py]: ~throw a 403~ raise :class:...HttpResponseError...

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_client.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_client.py]: sounds good, thanks!

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py (hunk 3 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py]: thanks!

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_polling.py (hunk 2 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_polling.py]: that makes a lot of sense, thanks!

**Pull request merged**